exoscale/api/compute.py

```diff
@@ -1035,6 +1052,185 @@ def _from_cs(cls, res):
         )
 
 
+@attr.s
+class InstancePool(Resource):
+    """
+    A Compute instance pool.
+
+    Attributes:
+        id (str): the instance pool unique identifier
+        name (str): the instance pool name
+        description (str): the instance pool description
+        zone (Zone): the zone in which the instance pool is located
+        size (int): the number of Compute instance members the pool manages
+        instance_type (InstanceType): the instance members type
+        instance_template (InstanceTemplate): the instance members template
+    """
+
+    compute = attr.ib(repr=False)
+    res = attr.ib(repr=False)
+    id = attr.ib()
+    name = attr.ib()
+    zone = attr.ib(repr=False)
+    size = attr.ib(repr=False)
+    instance_type = attr.ib(repr=False)
+    instance_template = attr.ib(repr=False)
+    description = attr.ib(default=None, repr=False)
+
+    @classmethod
+    def _from_cs(cls, compute, res, zone=None):
+        if zone is None:
+            zone = compute.get_zone(id=res["zoneid"])
+
+        return cls(
+            compute,
+            res,
+            id=res["id"],
+            name=res["name"],
+            description=res["description"],
+            zone=zone,
+            size=res["size"],
+            instance_type=compute.get_instance_type(id=res["serviceofferingid"]),
+            instance_template=compute.get_instance_template(zone, id=res["templateid"]),
+        )
+
+    @property
+    def instances(self):
+        """
+        Pool instance members.
+
+        Yields:
+            Instance: the next pool instance member
+
+        Note:
+            This property value is dynamically retrieved from the API, incurring extra
+            latency.
```

Comment:
If that is the case, then why tell the developer about the network call at all? It's an implementation detail.

Please either add enough information to be useful (how many API calls are made), or remove the implementation details altogether :)

Reply:
This mention is present on every single method/function performing API calls throughout the library as an indication for the developer to expect latency because of the network round-trip, how much latency is difficult to predict on paper :)